superset-frontend/src/explore/components/SaveModal.tsx

```diff
@@ -71,8 +84,11 @@ class SaveModal extends React.Component<SaveModalProps, SaveModalState> {
     this.state = {
       saveToDashboardId: null,
       newSliceName: props.sliceName,
+      datasetName: props.datasource?.name,
```

Comment:
It looks like this is where you want to set the default '' value? I am confused why you had the empty string in the interface

Reply:
yea i fixed that